test/junit/scala/collection/Sizes.scala

```diff
@@ -0,0 +1,46 @@
+package scala.collection
+
+import org.junit.Assert._
+import org.junit._
+import org.openjdk.jol.info.GraphLayout
+
+import scala.collection.mutable.ListBuffer
+
+object Sizes {
+  lazy val list:  Long = 24
+  lazy val listBuffer: Long = 32
+
+  def refArray(length:Int): Long = {
+    (16 + (length+1) * 4) /8 * 8
+  }
+  def wrappedRefArray(length:Int): Long = refArray(length) + 24
+
+}
```

Comment:
Where are you heading with this test? Is the intention to guard against accidental addition of `val`-s / `object`-s to core data structures?

That's okay, but I'd recommend using the `GraphLayout.subtract` idiom across the board to make an assertion about the net size of the data structure.

```scala
def netLayout(wrappper: Object, wrapped: Object): GraphLayout = GraphLayout.parseInstance(wrapper).substract(GraphLayout.parseInstance(wrapped)

def assertNetSize(expectedSize: Int, wrappper: Object, wrapped: Object): Unit = {
   val layout = netLayout(wrapper, wrapped)
   assertEquals(layout.toPrintable, expectedSize, layout.totalSize)
}
...

assertNetSize(24, someArray.toSeq, someArray))
```